sketch/tensor_full.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include "immintrin.h" // for AVX
+#include "nmmintrin.h" // for SSE4.2
+#include "sequence/alphabets.hpp"
+#include "sketch//sketch_base.hpp"
+#include "util/multivec.hpp"
+#include "util/timer.hpp"
+#include "util/utils.hpp"
+
+#include <cassert>
+#include <cmath>
+#include <iostream>
+#include <random>
+
+namespace ts { // ts = Tensor Sketch
+
+/**
+ * Computes tensor sketches for a given sequence as described in
+ * https://www.biorxiv.org/content/10.1101/2020.11.13.381814v1
+ * @tparam seq_type the type of elements in the sequences to be sketched.
+ */
+template <class seq_type>
+class FullTensor : public SketchBase<std::vector<double>, false> {
+  public:
+    // Tensor sketch output should be transformed if the command line flag is set.
+    constexpr static bool transform_sketches = false;
+
+    /**
+     * @param alphabet_size the number of elements in the alphabet S over which sequences are
+     * defined (e.g. 4 for DNA)
+     * @param normalize when true the counts will be normalized to relative frequencies with sum 1.
+     */
+    FullTensor(seq_type alphabet_size,
+               int32_t sequence_len,
```

Comment:
can this take negative values? if not, maybe make it uint32_t. Also, since this corresponds to the parameter `t`, I'd call it `subsequence_len` or simply `t` (`sequence_len` to me is the length of the actual sequence for which the tensor is computed)

Reply:
let it be noted that I have a strong dislike of unsigned types; they imply overflow semantics that we don't need/want, and because of that can be slower at times (probably not in our code though).
But I'll change it.